daffodil-test/src/test/resources/org/apache/daffodil/extensions/type_calc/inputTypeCalc.tdml

```diff
@@ -256,22 +236,6 @@
     </tdml:infoset>
   </tdml:unparserTestCase>
 
-  <tdml:unparserTestCase name="InputTypeCalc_unparse_keysetValue_03"
-    root="keysetValue_03" model="inputTypeCalc-Embedded.dfdl.xsd" description="Extensions - inputTypeCalc keysetValue transform">
```

Comment:
I have a question for @mbeckerle and @stevedlawrence.  Are there any other reasons for adding an unparserTestCase besides expecting an error on the unparse (the error case) or checking only the unparse (the roundTrip="none" case)?  An unparserTestCase is not allowed to check the twoPass / threePass cases, only the none / onePass cases, and I can't see any need for having a pair of identical parserTestCase and unparserTestCase tests both checking successful onePass cases.

In this TDML file, we should need only the parserTestCase tests because the tdml:testSuite has no defaultRoundTrip attribute and each test case has no roundTrip attribute.  The default behavior if nothing is specified in either the tdml:testSuite or the test case itself is onePass.   So, the parserTestCase tests parse the given data, compare to expected infoset, expect a match, and also unparse the new infoset, compare to given data, and expect a match.  The unparserTestCase tests just do the unparse, compare first followed by the parse, compare in the opposite order, and the order should be unimportant if you expect a successful onePass roundtrip.  Is my understanding correct?  

If so, most of the Daffodil TDML test cases should be parserTestCase tests (I count 3958 of them in Daffodil) and the need for unparserTestCase tests should be rare (I count 538 of them in Daffodil).

Reply:
> The default behavior if nothing is specified in either the tdml:testSuite or the test case itself is onePass. 

I think the default is actually "none":

https://github.com/apache/daffodil/blob/main/daffodil-tdml-lib/src/main/scala/org/apache/daffodil/tdml/RunnerFactory.scala#L92

So since the `Runner` and TDML files don't specify value, all the parserTestCases do not round trip and so we do need the unparserTestCases. We could potentially change the default for these files and remove those test unparseTestCase, but that would require some testing and confirmation that they are actually the same.

But your reasoning is all correct for onePass tests. In general, we should only need an unparserTestCase if roundTrip=onePass fails because the unparse doesn't exactly match the original data.

Reply:
Oh, I had read the statement "The default behavior if nothing is specified in either the tdml:testSuite or the test case itself is onePass" in the last section, Round-Trip Testing, of the Daffodil website's [Test Data Markup Language (TDML)](https://daffodil.apache.org/tdml/) documentation.  Since that statement and the source code disagree with each other, we need to fix one of them.  I assume changing defaultRoundTripDefaultDefault in the source code will break lots of TDML tests so I will open a JIRA issue on the documentation page.